src/components/CardSelector/CardSelector.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import clsx from 'clsx';
+
+import './CardSelector.scss';
+import type {CardSelectorProps} from './CardSelector.types';
+import {Typography} from '~/components';
+import {FileBroken, HandleDrag, Image} from '~/icons/components';
+
+export const CardSelector = React.forwardRef<HTMLDivElement, CardSelectorProps>(({
+    displayName,
+    systemName,
+    chips,
+    information,
+    thumbnailURL,
+    thumbnailType = 'preview',
+    thumbnailAlt,
+    id,
+    className,
+    isDraggable = false,
+    isDisabled = false,
+    isReadOnly = false,
+    cardAction,
+    hasError = false,
+    errorMessage,
+    onClick = () => undefined,
+    ...props
+}, ref) => {
+    const classNameProps = clsx(
+        'moonstone-cardSelector',
+        (isDisabled || isReadOnly) && 'moonstone-cardSelector_disabled',
+        'flexFluid',
+        'flexRow_nowrap',
+        'alignCenter',
+        className
+    );
+
+    const handleOnClick: React.MouseEventHandler = e => {
+        if (isReadOnly || isDisabled) {
+            return;
+        }
+
+        onClick(e);
+        (e.currentTarget as HTMLElement).blur();
+    };
+
+    if (hasError) {
+        return (
+            <div
+                ref={ref}
+                id={id}
+                className={clsx('moonstone-cardSelector_error', 'flexRow_center', 'alignCenter')}
+                {...props}
```

Comment:
We also must handle `disabled` and `readonly` when the `CardSelector` is on error